zerver/webhooks/dialogflow/view.py

```diff
@@ -0,0 +1,34 @@
+# Webhooks for external integrations.
+from typing import Text, Any, Dict
+from django.http import HttpRequest, HttpResponse
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.actions import check_send_private_message
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.models import UserProfile, get_user_profile_by_email
+
+@api_key_only_webhook_view("dialogflow")
+@has_request_variables
+def api_dialogflow_webhook(request: HttpRequest, user_profile: UserProfile,
+                           payload: Dict[str, Any]=REQ(argument_type='body'),
+                           email: str=REQ(default='foo')) -> HttpResponse:
+    status = payload["status"]["code"]
+
+    if status == 200:
+        result = payload["result"]["fulfillment"]["speech"]
+        if result == '':
+            alternate_result = payload["alternateResult"]["fulfillment"]["speech"]
+            if alternate_result == '':
+                output = u"Sorry. I didn't get that."
+            else:
+                output = alternate_result
+        else:
+            output = result
+    else:
+        error_status = payload["status"]["errorDetails"]
+        output = u"{} - {}".format(status, error_status)
+
+    profile = get_user_profile_by_email(email)
+    body = u"Response : **{}**".format(output)
```

Comment:
Okay, so you don't need to say `Response :`. It is already clear that the bot is responding to you in a private chat so this is redundant. Also, don't bold the response! I would just recommend passing in `output` directly to `check_send_private_message`. :)